x/local-proxy/proxy/proxy.go

```diff
@@ -0,0 +1,13 @@
+package proxy
+
+type Proxy interface {
+	// GetAddr returns the address of the proxy server if it is configured to listen on a specific address.
+	// If address is not configured and the server is not started, it returns an empty string.
+	GetAddr() string
+	// StartServer starts the proxy server. If no address is specified, it will listen on a random localhost port
+	StartServer(addr ...string) error
```

Comment:
Why would it take a list?

Reply:
I wouldn't but it allows you to call it without the address `StartServer()`.
I'm not sure how that works with other languages when they use Go as bindings. Probably it doesn't make much sense and ambiguous, but could be convenient. Do you think we should change it?

Reply:
Let's take a single string then. The convention is to pass port "0" or an empty port if you need the system to assign you one. See https://pkg.go.dev/net#Listen

```suggestion
	StartServer(addr string) error
```